src/controllers/words.js

```diff
@@ -107,6 +113,72 @@ export const getWords = async (req, res, next) => {
   }
 };
 
+/* Gets words from MongoDB by the searched keywords and specified wordClass */
+export const getWordsFilteredByWordClass =  async (req, res, next) =>{
+  try {
+    const wordClass = parseWordClass(req.params.wordClass);
+    const hasQuotes = req.query.keyword && (req.query.keyword.match(/["'].*["']/) !== null);
+    if (hasQuotes) {
+      req.query.keyword = req.query.keyword.replace(/["']/g, '');
+    }
+    const {
```

Comment:
The main thing that we want to avoid is duplicating code. It looks like you copied this logic from the `getWords` function, which makes sense to do, but this makes it harder for the codebase to be maintained since we are duplicating lines of code that could be placed in a separate function.

So you're on the right track of thinking about reusing the logic found in `getWords` but rather than copying and pasting that logic into this function, I think you should move this logic out to a separate function that can be called in both `getWords` and `getWordsFilteredByWordClass`